modules/custom/activity_creator/src/ActivityNotifications.php

```diff
@@ -100,38 +125,36 @@ public function markEntityNotificationsAsRead(AccountInterface $account, EntityB
    * @param \Drupal\Core\Entity\EntityBase $entity
    *   Entity object.
    *
-   * @throws \Drupal\Core\Entity\EntityStorageException
+   * @deprecated in opensocial:8.x-7.0 and is removed from opensocial:8.x-7.1. Use
+   *   \Drupal\activity_creator\ActivityNotifications
+   * ::markEntityNotificationsAsRead() instead.
+   *
+   * TODO: Change @see to point to a change record.
+   * @see https://www.drupal.org/project/social/issues/3087083
    */
   public function markEntityAsRead(AccountInterface $account, EntityBase $entity) {
 
     // Retrieve all the activities referring this entity for this account.
     $ids = $this->getNotificationIds($account, [ACTIVITY_STATUS_RECEIVED, ACTIVITY_STATUS_SEEN], $entity);
-
-    $activities = Activity::loadMultiple($ids);
-    foreach ($activities as $activity) {
-      $this->changeStatusOfActivity($activity, ACTIVITY_STATUS_READ);
-    }
+    $this->changeStatusOfActivity($ids, ACTIVITY_STATUS_READ);
 
   }
 
   /**
    * Change the status of an activity.
    *
-   * @param \Drupal\activity_creator\Entity\Activity $activity
-   *   Activity object.
+   * @param array $ids
+   *   Array of IDs.
    * @param int $status
    *   See: activity_creator_field_activity_status_allowed_values()
-   *
-   * @return bool|int
-   *   SAVED_NEW or SAVED_UPDATED is returned depending on the operation
-   *   performed.
-   *
-   * @throws \Drupal\Core\Entity\EntityStorageException
    */
-  public function changeStatusOfActivity(Activity $activity, $status = ACTIVITY_STATUS_RECEIVED) {
-    $activity->set('field_activity_status', $status);
-
-    return $activity->save();
+  public function changeStatusOfActivity(array $ids, $status = ACTIVITY_STATUS_RECEIVED) {
```

Comment:
We may want to put this update in a try catch.

In the catch we could log our failed attempt.

Reply:
Additionally you may want to check for an empty array of `$ids` here.